src/snarl_seed_clusterer.hpp

```diff
@@ -381,59 +453,68 @@ class NewSnarlSeedClusterer {
+        //child_range_start/end are the iterators to the start (inclusive) and end (exclusive) 
+        //of range of the snarl in snarl_to_children
         void cluster_one_snarl(TreeState& tree_state, size_t snarl_clusters_index, 
                  std::multimap<size_t, size_t>::iterator child_range_start, std::multimap<size_t, size_t>::iterator child_range_end) const;
 
-        //Cluster the seeds in a chain given by chain_index_i, an index into
+        //Cluster the seeds in a chain given by chain_clusters_index, an index into
         //distance_index.chain_indexes
-        //If the depth is 0, also incorporate the top-level seeds from tree_state.top_level_seed_clusters
-        //Chain children are tuples<net_handle, (child index, inf) or (seed read num, seed index), chain component, prefix sum>
+        //chain_range_start/end are iterators to the start (inclusive) and end (exclusive) of the
+        //range in chain_to_children representing children of this chain
+        //The range must be ordered
+        //
         //If the children of the chain are only seeds on nodes, then cluster as if it is a node
-        void cluster_one_chain(TreeState& tree_state, size_t chain_clusters_index, vector<tuple<net_handle_t, size_t, size_t, size_t, size_t>>& children_in_chain, bool only_seeds, bool is_top_level_chain) const;
+        void cluster_one_chain(TreeState& tree_state, size_t chain_clusters_index, 
+            const std::vector<ParentToChildMap::ParentChildValues>::iterator& chain_range_start,
+            const std::vector<ParentToChildMap::ParentChildValues>::iterator& chain_range_end,
+            bool only_seeds, bool is_top_level_chain) const;
 
         //Helper function for adding the next seed to the chain clusters
         void add_seed_to_chain_clusters(TreeState& tree_state, NodeClusters& chain_clusters,
-                                        std::tuple<net_handle_t, size_t, size_t, size_t, size_t>& last_child, net_handle_t& last_child_handle, 
+                                        ParentToChildMap::ParentChildValues& last_child,
                                         size_t& last_prefix_sum, size_t& last_length, size_t& last_chain_component_end, 
-                                        vector<pair<pair<size_t, size_t>, pair<size_t, size_t>>>& cluster_heads_to_add_again,
-                                        bool& found_first_node, vector<bool>& found_first_node_by_read,
-                                        tuple<net_handle_t, size_t, size_t, size_t, size_t>& current_child_indices, bool is_first_child, bool is_last_child,
+                                        vector<ClusterIndices>& cluster_heads_to_add_again,
+                                        bool& found_first_node, pair<bool, bool>& found_first_node_by_read,
+                                        const ParentToChildMap::ParentChildValues& current_child, bool is_first_child, bool is_last_child,
```

Comment:
I feel like this should really be some kind of identifier (`net_handle_t`? Reference to record for the child but with a better type name?) for the child, and then we should just ask something (the `TreeState`?) if it is first/last in its parent.

Also `last_child` above is really `prev_child`, right? Whereas here "last" is really rightmost?